more arguments for prepare

thebookins opened the pull request:
To use the `prepare` step in an `openaps` workflow, and properly resolve pump suspends, the current clock time and duration of the history window need to be passed in as arguments and routed to the `clean` step. This PR adds options `--end`, `--start` and `--duration` to the interface for `prepare`. For consistency, the `clean` step is also modified to match `trim` and `prepare` and will now accept start and end times in JSON format. Needed for https://github.com/openaps/oref0/pull/279.

loudnate commented on openapscontrib/mmhistorytools/historytools.py:
```diff
@@ -114,6 +114,11 @@ def __init__(self, trimmed_history, start_datetime=None, end_datetime=None):
         self.end_datetime = end_datetime
 
         if len(trimmed_history) > 0:
+            if start_datetime is None and end_datetime is not None and duration_hours is not None:
```
Would it be worthwhile to add test coverage for the duration argument? You should be able to copy/adapt the existing tests for duration.

thebookins replied:
Good idea, I'll add test coverage.

thebookins commented:
I've added test coverage for the duration argument, and found a mistake in the process (now fixed). Thanks @loudnate for your feedback, and for building this great set of tools in the first place.

thebookins commented:
Thanks. Not expecting to make any more changes, so would be great if you could go ahead with the release.
